File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Hyperlinks/addhyperlinktoshape.rb
module Asposediagramjava
  module AddHyperlinkToShape
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Initialize Hyperlink object
        hyperlink = Rjb::import('com.aspose.diagram.Hyperlink').new
        
        # Set address value
        hyperlink.getAddress().setValue("http://www.google.com/")
        
        # Set sub address value
        hyperlink.getSubAddress().setValue("Sub address here")
        
        # Set description value
        hyperlink.getDescription().setValue("Description here")
        
        # Set name
        hyperlink.setName("MyHyperLink")

        # Add hyperlink to the shape
        diagram.getPages().getPage("Flow 1").getShapes().getShape(1).getHyperlinks().add(hyperlink)

        # Save diagram
        diagram.save(data_dir + "Hyperlinks.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Added hyperlik to shape successfully!"
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithHyperlinks/AddHyperlinkToShape.php
<?php
namespace Aspose\Diagram\WorkingwithHyperlinks;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\Hyperlink as Hyperlink;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class AddHyperlinkToShape{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir . "Drawing.vsd");

        # Initialize Hyperlink object
        $hyperlink = new Hyperlink();

        # Set address value
        $hyperlink->getAddress()->setValue("http://www.google.com/");

        # Set sub address value
        $hyperlink->getSubAddress()->setValue("Sub address here");

        # Set description value
        $hyperlink->getDescription()->setValue("Description here");

        # Set name
        $hyperlink->setName("MyHyperLink");

        # Add hyperlink to the shape
        $diagram->getPages()->getPage("Flow 1")->getShapes()->getShape(1)->getHyperlinks()->add($hyperlink);

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir . "Hyperlinks.vdx", $saveFileFormat->VDX);

        print "Added hyperlik to shape successfully!".PHP_EOL;
    }
}